src/buildblock/ProjDataInfoSubsetByView.cxx

```diff
@@ -0,0 +1,265 @@
+{
+  this->org_proj_data_info_sptr->set_max_tangential_pos_num(max_tang_poss);
+  // currently need to do this
+  ProjDataInfo::set_max_tangential_pos_num(max_tang_poss);
+}
+
+float ProjDataInfoSubsetByView::get_tantheta(const Bin& bin) const
+{
+  return this->org_proj_data_info_sptr->get_tantheta(get_original_bin(bin));
+}
+
+float ProjDataInfoSubsetByView::get_phi(const Bin& bin) const
+{
+  return this->org_proj_data_info_sptr->get_phi(get_original_bin(bin));
+}
+
+float ProjDataInfoSubsetByView::get_t(const Bin& bin) const
+{
+  return this->org_proj_data_info_sptr->get_t(get_original_bin(bin));
+}
+
+float ProjDataInfoSubsetByView::get_m(const Bin& bin) const
+{
+  return this->org_proj_data_info_sptr->get_m(get_original_bin(bin));
+}
+
+float ProjDataInfoSubsetByView::get_s(const Bin& bin) const
+{
+  return this->org_proj_data_info_sptr->get_s(get_original_bin(bin));
+}
+
+void ProjDataInfoSubsetByView::get_LOR(LORInAxialAndNoArcCorrSinogramCoordinates<float>& lor,
+                                       const Bin& bin) const
+{
+  return this->org_proj_data_info_sptr->get_LOR(lor, get_original_bin(bin));
+}
+
+float ProjDataInfoSubsetByView::get_sampling_in_t(const Bin& bin) const
+{
+  return this->org_proj_data_info_sptr->get_sampling_in_t(get_original_bin(bin));
+}
+
+float ProjDataInfoSubsetByView::get_sampling_in_m(const Bin& bin) const
+{
+  return this->org_proj_data_info_sptr->get_sampling_in_m(get_original_bin(bin));
+}
+
+float ProjDataInfoSubsetByView::get_sampling_in_s(const Bin& bin) const
+{
+  return this->org_proj_data_info_sptr->get_sampling_in_s(get_original_bin(bin));
+}
+
+Bin ProjDataInfoSubsetByView::get_bin(const LOR<float>& lor) const
+{
+  return get_bin_from_original(this->org_proj_data_info_sptr->get_bin(lor));
+}
+
+bool ProjDataInfoSubsetByView::contains_full_data() const
+{
+  // Warning: this only checks size, so assumes no double-ups of views
+  return view_to_org_view_num.size() == static_cast<std::size_t>(org_proj_data_info_sptr->get_num_views());
+}
+
+bool ProjDataInfoSubsetByView::operator>=(const ProjDataInfo& proj) const
+{
+   if (typeid(*this) != typeid(proj))
+     {
+       if (this->contains_full_data())
+         return (*this->org_proj_data_info_sptr) >= proj;
+       else
+         return false;
+     }
+
+   auto smaller_proj_data_info = static_cast<const ProjDataInfoSubsetByView&>(proj);
+
+   if (!((*this->org_proj_data_info_sptr) >= (*smaller_proj_data_info.org_proj_data_info_sptr)))
+    { return false; }
+
+  // check all of smaller_proj_data_info org_views are in this subset
+  for (int view_num_idx = 0; view_num_idx < smaller_proj_data_info.get_num_views(); ++view_num_idx)
+  {
+    int org_view_num = smaller_proj_data_info.view_to_org_view_num[view_num_idx];
+    if (std::find(this->view_to_org_view_num.begin(), this->view_to_org_view_num.end(), org_view_num)
+          == this->view_to_org_view_num.end())
+    { return false; }
+  }
+
+  return base_type::operator>=(proj);
```

Comment:
So the issue is that the `base_type::operator>=(proj)` returns False when we want True because it works by reducing the size of `this` until it should equal that of `proj`, and then compares for equality.
We need to do something like this:
```c++
  // base_type::operator>= works by reducing the larger ProjDataInfo (this) until they're equal.
  // We'll need to, therefore, compare with a subset of this that only includes the views of
  // smaller_proj_data_info
  std::vector<int> smaller_proj_data_view_nums;
  for (auto org_view_num : smaller_proj_data_info.view_to_org_view_num)
    smaller_proj_data_view_nums.push_back(this->org_view_to_view_num[org_view_num]);
  ProjDataInfoSubsetByView this_projdata_info_with_other_views(
    this->create_shared_clone(), smaller_proj_data_view_nums);

  // I need to call something like:
  base_type::operator>=(this_projdata_info_with_other_views, smaller_proj_data_info);
```
How can I call `ProjDataInfo::operator>=` here? In Python I would do like `ProjDataInfo.operator>=(this_projdata_info_with_other_views, smaller_proj_data_info)` because `self`/`this` is passed around as an argument. Can I do this?

Reply:
An alternative is to have a non-overridden function in `ProjDataInfo`, like `ProjDataInfo::is_superset`, than can be accessed, and have `ProjDataInfo::operator>=` call it.

Reply:
You can call `base_type::operator>=` by the way, no problem. But I can't see how that'd help.  In the end, we have to compare if the views are there, and if the `org` is also larger.. Can you be more explicit on which test fails and why? to me the `if (typeid(*this) != typeid(proj)) ... else`  statement seems wrong.

Reply:
> You wrote this, so I tried to guess intent here.

😄 . Ok, I understand my own logic again!

> I think `ProjDataInfoSubsetByView::operator>=` should check that anything subset-specific is equal, and then use `ProjDataInfo::operator>=` to check the rest.

well, we have above all this `if (!((*this->org_proj_data_info_sptr) >= (*smaller_proj_data_info.org_proj_data_info_sptr)))...`, so it's doing the check in the other order, but that seems fine to me.

So why do we not simply return `true` after the check on `views` ? I'm missing something I guess.

In any case, I don't like the introduction of `is_superset` too much. We now have a virtual function `operator >=` calling another virtual `is_superset`. We override `operator>=`, but not `is_superset`, which from a normal developer perspective is very weird. Hopefully we don't need `is_superset` therefore...

Reply:
Yeah I agree it is quite ugly.

But I see your argument now as to whether we need to do any check. I wanted to confirm segments, tangential_pos, axial_pos, etc. were `<=` for the subset. But since we already check these for the `org_proj_data_info` and every time we update for the subset we update for the original, that should be covered.

There is one technicality to bypass this check: if  the `org_proj_data` was modified without the subset info. But I don't think this can happen as outside the class you can only access a readonly pointer to `org_proj_data`. We just need to be careful within the class: I just noticed that `set_num_tangential_poss` doesn't update both subset and original.

Reply:
Removed `is_subset` and fixed `set_num_tangential_poss`